Using string_view on config/xml/utils

marzojr opened the pull request:
Not fully using everywhere, but enough to make a good start.

marzojr committed:
Using string_view on config/xml/utils

Not fully using everywhere, but enough to make a good start.

wench commented:
We already have some case insensitive string comparison functions for null terminated strings in istring.h/cc, you might want to put iequals and safe_tolower  in there

marzojr committed:
Moving iequals to istring.{cc,h}

marzojr committed:
Unused variable

marzojr committed:
Adding include for static analyzers

marzojr committed:
More string_view

- Removing redundant uses of c_str()
- Changed a few mistaken uses of strcasecmp to Pentagram::iequals
- Changed (almost) all cases of Pentagram::strn?casecmp to Pentagram::iequals

marzojr committed:
Fixing static analyzers on headers

marzojr committed:
Fixing final few redundant calls to c_str()

marzojr committed:
Fixing up codeql stuff

marzojr commented:
So I have been working on this a bit. I have one absolutely giant commit which I am not sure I can make smaller because it is a mess of interconnected changes.

And I am thinking that maybe I want to work on adding cmake support first, with the intent of using [fmtlib](https://github.com/fmtlib/fmt), [scnlib](https://github.com/eliaskosunen/scnlib), and maybe [spdlog](https://github.com/gabime/spdlog/). They would all help improve some of the things I am working on on both this pull request, and on #477.

Another issue is the overlap of this and #477. Both make several similar changes on similar lines, and merging these is going to be a pain.

So I am thinking of creating a dev branch that (1) adds cmake support, (2) starts using fmtlib and scnlib in places where snprintf/scanf/etc are used, and (3) merge this pull and #477 onto said branch to work.

That said, I think I will add a commit to this that changes the way doubles are read from/written to config because the current method has issues with localization.

DominusExult commented:
That's a sound way to move forward with this IMO
